src/Uno.UI/UI/Xaml/Window.cs

```diff
@@ -255,19 +277,12 @@ internal void OnActivated(CoreWindowActivationState state)
 
 		internal void OnVisibilityChanged(bool newVisibility)
 		{
-			if (Visible != newVisibility)
+			if (!_wasEverCodeActivated)
 			{
-				if (this.Log().IsEnabled(LogLevel.Debug))
-				{
-					this.Log().LogDebug($"Window visibility changing to {newVisibility}");
-				}
-
-				Visible = newVisibility;
-
-				var args = new VisibilityChangedEventArgs() { Visible = newVisibility };
-				CoreWindow.OnVisibilityChanged(args);
-				VisibilityChanged?.Invoke(this, args);
+				return;
 			}
```

Comment:
It may be interesting to keep the debug message ?

Reply:
The log is still there, just moved to the property setter 👍